sdk/core/System.ClientModel/src/Internal/BitVector640.cs

```diff
@@ -28,136 +28,66 @@ internal struct BitVector640
     private ulong _bits8;
     private ulong _bits9;
 
-    public bool this[int i]
+    // To get or set the correct bit, we need to find a bracket and an offset inside bracket
+    // For that, we have GetBracket and GetOffset methods.
+    // ===================================================
+    // Example: find the bracket and the offset for index 621
+    // In binary form, 621 is 0b1001101101.
+    // First 4 bits represent the bracket number, while the last 6 bits represent the offset in the bracket
+    // (ulong is 64-bit type, hence offset is in the [0, 63] range)
+    //
+    //    9 - bracket number
+    // ╔══╩══╗
+    // 1 0 0 1 1 0 1 1 0 1
+    //         ╚════╦════╝
+    //              45 - offset in the bracket
+    // ╔════════════╩════════════════════════════════════════════════════════╗
+    // 00000000 00000000 00100000 00000000 00000000 00000000 00000000 00000000
+    //                     ↑
+    //      bit #45 in 64-bit mask (indexing starts with 0)
+    //
+    // To get the bracket number, we right shift the index by 6
+    //    (6 bits required to represent value in [0, 63] range)
+    // To get the offset, we apply the 0b111111 mask to the index to get value from the last 6 bits,
+    //    then shift left a single bit by that value to get mask for the bracket,
+    //    then apply mask to the bracket
+
+    private static ref ulong GetBracket(ref BitVector640 vector, int index)
     {
-        readonly get
+        switch (index >> 6)
         {
-            // "Index" of the ulong the bit is stored in.
-            int index = i >> 6;
-
-            // i % 6, i.e. the offset of the bit in the ulong.
-            int mod = i & 0b111111;
-
-            // A mask that lets us access the single bit
-            ulong mask = 1ul << mod;
-
-            // The storage ulong and the mask
-            ulong bit = Get(index) & mask;
-
-            // If the bit equals the mask, the bit was set to true.
-            return bit == mask;
-        }
-        set
-        {
-            // "Index" of the ulong the bit is stored in.
-            int index = i >> 6;
-
-            // i % 6, i.e. the offset of the bit in the ulong.
-            int mod = i & 0b111111;
-
-            // A mask that lets us access the single bit
-            ulong mask = 1ul << mod;
-
-            // Set the bit in question to the passed-in value
-            Set(index, mask, value);
+            case 0: return ref vector._bits0;
```

Comment:
Are you sure this is faster? Do you have a perf test?

Reply:
I would also like to see evidence that this improves performance.

Reply:
Added benchmark above

Reply:
Why is it faster?  Is it easy to explain?  Is it just that you added [AggressiveInlining] or is there something about how you modified the algorithm?

Reply:
One observation with regard to perf is that branch statements are expensive processor operations.  It would be cool if we could find a way to rewrite this that doesn't use branching at all (the switch statement translates to branch statements), but I didn't spend the time to push it this far.

Reply:
There is no branching in this particular case, but we can make it even faster (see third row above):
```cs
[MethodImpl(MethodImplOptions.AggressiveInlining)]
private static ref ulong GetBracket(ref BitVector640_Unsafe vector, int index)
    => ref Unsafe.Add(ref Unsafe.As<BitVector640_Unsafe, ulong>(ref vector), index >> 6);
```

Reply:
yes. thanks. Looks good